test/e2e/scripts/prow_run_smoke_test.sh

```diff
@@ -0,0 +1,165 @@
+    fi
+    
+    # Check if we're logged in
+    if ! oc whoami >/dev/null 2>&1; then
+        echo "❌ ERROR: Not logged into OpenShift. Please run 'oc login' first"
+        exit 1
+    fi
+    
+    echo "✅ Prerequisites met - logged in as: $(oc whoami)"
+}
+
+deploy_maas_platform() {
+    echo "Deploying MaaS platform on OpenShift..."
+    
+    if [ ! -f "$PROJECT_ROOT/deployment/scripts/deploy-openshift.sh" ]; then
+        echo "❌ ERROR: Deployment script not found: $PROJECT_ROOT/deployment/scripts/deploy-openshift.sh"
+        exit 1
+    fi
+    
+    if ! "$PROJECT_ROOT/deployment/scripts/deploy-openshift.sh"; then
+        echo "❌ ERROR: MaaS platform deployment failed"
+        exit 1
+    fi
+    
+    echo "✅ MaaS platform deployment completed"
+}
+
+deploy_models() {
+    echo "Deploying simulator Model"
+    if ! (cd "$PROJECT_ROOT" && kustomize build docs/samples/models/simulator/ | kubectl apply -f -); then
+        echo "❌ ERROR: Failed to deploy simulator model"
+        exit 1
+    fi
+    echo "✅ Simulator model deployed"
+    
+    echo "Waiting for model to be ready..."
+    sleep 30
+    echo "✅ Simulator Model deployed"
+}
+
+validate_deployment() {
+    echo "Deployment Validation"
+    if [ "$SKIP_VALIDATION" = false ]; then
+        if ! "$PROJECT_ROOT/deployment/scripts/validate-deployment.sh"; then
+            echo "❌ ERROR: Deployment validation failed"
+            exit 1
+        fi
+        echo "✅ Deployment validation completed"
+    else
+        echo "⏭️  Skipping validation"
+    fi
+}
+
+setup_maas_users() {
+    echo "Setting up Maas users for testing"
+    if ! (cd "$PROJECT_ROOT" && bash test/e2e/scripts/setup_maas_users_openshift.sh); then
+        echo "❌ ERROR: Failed to setup Maas users on OpenShift"
+        exit 1
+    fi
+    
+    # Source the environment variables created by the setup script
+    local env_file="$PROJECT_ROOT/maas-users.env"
+    if [ -f "$env_file" ]; then
+        source "$env_file"
+    else
+        echo "❌ ERROR: Users credentials not found"
+        exit 1
+    fi
+    
+    echo "✅ Maas users setup completed"
+}
+
+login_as_user() {
+    echo "Logging in as user: $1"
+    if ! (oc login -u "$1" -p "$2"); then
+        echo "❌ ERROR: Failed to login as user: $1"
+        exit 1
+    fi
+    echo "✅ User: $1 logged in successfully as: $(oc whoami)"
+}
+
+setup_vars_for_tests() {
+    echo "-- Setting up variables for tests --"
+    CLUSTER_DOMAIN="$(oc get ingresses.config.openshift.io cluster -o jsonpath='{.spec.domain}')"
+    export CLUSTER_DOMAIN
+    if [ -z "$CLUSTER_DOMAIN" ]; then
+        echo "❌ ERROR: Failed to retrieve OpenShift cluster domain"
+        exit 1
+    fi
+    
+    export HOST="maas.${CLUSTER_DOMAIN}"
+    export MAAS_API_BASE_URL="http://${HOST}/maas-api"
+
+    echo "CLUSTER_DOMAIN: ${CLUSTER_DOMAIN}"
+    echo "HOST: ${HOST}"
+    echo "MAAS_API_BASE_URL: ${MAAS_API_BASE_URL}"
+
+    echo "✅ Variables for tests setup completed"
+}
```

Comment:
Are all variables expected by the scripts we call down the line or can we rely on a fact that some of them will be constructed/autodiscovered?